openeo-geotrellis/src/main/scala/org/openeo/geotrellis/netcdf/NetCDFRDDWriter.scala

```diff
@@ -801,6 +888,120 @@ object NetCDFRDDWriter {
     if (bandsMetadata.containsKey("OFFSET")) netcdfFile.addVariableAttribute(variableName,"add_offset",bandsMetadata.get("OFFSET").toFloat)
   }
 
+  private def setupAssetMetadata[K: SpatialComponent : Boundable : ClassTag](metadata: TileLayerMetadata[K], dates: List[Int], bandNames: ArrayList[String], addBandsStats: Boolean, histograms:scala.collection.mutable.Map[String,(StreamingHistogram,Int)]): java.util.Map[String, Any] = {
+    val assetMetadata = if (dates.nonEmpty) {
+      new util.HashMap[String,Any](util.Map.of("time", new util.HashMap[String,Any](util.Map.of("type", "temporal", "extent",Array(dates.head, dates.last), "values", dates.toArray))))
+    } else new java.util.HashMap[String,Any]()
+    val bands = if (addBandsStats) {
+      val map = new util.ArrayList[util.Map[String,Any]]()
+      histograms.foreach {case (name,histogramAndSize) => {
+        val statistics = histogramAndSize._1.statistics()
+        val mapStatistics = if (statistics.isDefined) {
+          new util.HashMap[String, Any](util.Map.of("max", statistics.get.zmax, "min", statistics.get.zmin, "mean", statistics.get.mean,"stddev",statistics.get.stddev, "valid_percent", statistics.get.dataCells.toDouble/histogramAndSize._2))
+        } else new util.HashMap[String, Any](util.Map.of("valid_percent", 0.0))
+        val band = new util.HashMap[String,Any](util.Map.of("name", name, "statistics", mapStatistics))
+        map.add(band)
+
+      }}
+      map
```

Comment:
```suggestion
      val maps = new util.ArrayList[util.Map[String,Any]]()
      bandHistograms.foreach {case (bandName, (histogram, size)) =>
        val statistics = histogram.statistics()
        val mapStatistics = statistics.fold(util.Map.of("valid_percent", 0.0)) { statistics =>
          util.Map.of("max", statistics.zmax, "min", statistics.zmin, "mean", statistics.mean,"stddev",statistics.stddev, "valid_percent", statistics.dataCells.toDouble/size)
        }
        val band = util.Map.of("name", bandName, "statistics", mapStatistics: Any)
        maps.add(band)
      }
      maps
```

Improve naming: `maps` is still not ideal but at least it shows that it is a list of maps rather than a single map.

`Option` is a collection (of 0 or 1 elements) so you can apply typical operations like `map`, `filter` and `fold` to it; in this case `fold` avoids calls to `isDefined` and `get`.